plugins/modules/site_intent.py

```diff
@@ -774,10 +783,47 @@ def get_diff_deleted(self, config):
         site_exists = self.have.get("site_exists")
 
         if site_exists:
+            # Check here if the site have the childs then fetch it using get membership API and then sort it
+            # in reverse order and start deleting from bottom to top
+            site_id = self.have.get("site_id")
+            mem_response = self.dnac._exec(
+                family="sites",
+                function="get_membership",
+                params={"site_id": site_id},
+            )
+            site_response = mem_response.get("site").get("response")
+
+            if len(site_response) > 0:
+                # Sorting the response in reverse order based on hierarchy levels
```

Comment:
What happens if the site_response is lessthan or equal to 0. Can we return by setting self.result and log a message at here itself?

```
if len(site_response) <= 0:
 ..
 ..
 ..
```
So we don't need to write more statements inside this if block.

Reply:
added